src/test/regress/bin/normalize.sed

```diff
@@ -100,6 +100,14 @@ s/of relation ".*" violates not-null constraint/violates not-null constraint/g
 s/partition ".*" would be violated by some row/partition would be violated by some row/g
 s/of relation ".*" contains null values/contains null values/g
 
+# libpq message changes
+# We ignore multiline error messages, and substitute first line with a single line
+# alternative that is used in some older libpq versions.
+s/(ERROR: |WARNING: |error:) server closed the connection unexpectedly/\1 connection not open/g
+/^\s*This probably means the server terminated abnormally$/d
+/^\s*before or while processing the request.$/d
+/^\s*connection not open$/d
```

Comment:
We seem to be matching all lines separately, have you considered this https://unix.stackexchange.com/a/26290 to match mutli line messages.

We could use this to only match the `connection not opened` messages that changed.

Causing hopefully less other output files to be changed.

Reply:
If we can do that, +1

Reply:
Writing a multiline sed script is totally doable. However our `diff-filter` script does not recognize those rules and we end up with a strange `regression.diffs` file.

In `diff-filter` we parse the lines of `normalize.sed` one by one, and this is not a trivial thing that I can easily fix now.

https://github.com/citusdata/citus/blob/67ac3da2b016bf36e08dd7b13e03ebb21984207e/src/test/regress/bin/diff-filter#L30-L37

For example, I came up with an alternative solution on another branch https://github.com/citusdata/citus/compare/alternative-libpq-fix that looks fine. However our diff-filter scripts generate some rather bizarre regression.diffs outputs.

Here is a the rule in the branch that generates a fairly reasonable code change when applied to test outputs:

```sed
/(ERROR: |WARNING: |error:) server closed the connection unexpectedly/ {
	N; s/\s*This probably means the server terminated abnormally//g
	N; s/\s*before or while processing the request.//g
	N; s/\s*connection not open//g
}

s/(ERROR: |WARNING: |error:) server closed the connection unexpectedly/\1 connection not open/g
```